bundles/org.eclipse.swt/Eclipse SWT Tests/win32/org/eclipse/swt/widgets/DisplayWin32Test.java

```diff
@@ -8,48 +8,89 @@
 
 public class DisplayWin32Test {
 
-	private Display display;
+	private boolean initialMonitorSpecificScaling;
 
 	@BeforeAll
 	public static void assumeIsFittingPlatform() {
 		PlatformSpecificExecution.assumeIsFittingPlatform();
 	}
 
 	@BeforeEach
-	public void createDisplay() {
-		display = new Display();
+	public void storeState() {
+		initialMonitorSpecificScaling = DPIUtil.isMonitorSpecificScalingActive();
 	}
 
 	@AfterEach
-	public void destroyDisplay() {
-		display.dispose();
+	public void restoreState() {
+		DPIUtil.setMonitorSpecificScaling(initialMonitorSpecificScaling);
 	}
 
 	@Test
+	public void monitorSpecificScaling_activate() {
+		DPIUtil.setMonitorSpecificScaling(true);
+		Display display = new Display();
+		try {
+			assertTrue(display.isRescalingAtRuntime());
+			assertEquals(OS.DPI_AWARENESS_CONTEXT_PER_MONITOR_AWARE_V2, OS.GetThreadDpiAwarenessContext());
+		} finally {
+			display.dispose();
+		}
+	}
+
+	@Test
+	public void monitorSpecificScaling_deactivate() {
+		DPIUtil.setMonitorSpecificScaling(false);
+		Display display = new Display();
+		try {
+			assertFalse(display.isRescalingAtRuntime());
```

Comment:
Is it unnecessary/incorrect to assert that the scaling is set to `DPI_AWARENESS_CONTEXT_SYSTEM_AWARE` just like the "old" test `setRescaleAtRuntime_deactivate` does?

```suggestion
			assertFalse(display.isRescalingAtRuntime());
			assertEquals(OS.DPI_AWARENESS_CONTEXT_SYSTEM_AWARE, OS.GetThreadDpiAwarenessContext());
```

This asymmetry in the tests looks confusing to me.

Reply:
Yes, it is incorrect.

Reply:
Care to elaborate? I'm missing some pieces of the puzzle here.

Reply:
The application runs in "default" mode in this test case, so the actual DPI awareness depends on what the JVM is configured to use as process-wide DPI awareness. That's why assuming a specific DPI awareness is incorrect.

Reply:
Ah, the missing piece (in my mind) was the fact that `Display::setRescalingAtRuntime` actively overrides the system default both when passing `true` and `false` as a parameter while `DPIUtil::setMonitorSpecificScaling` only overrides the system value when passing `true` as a parameter. 
The problem I see here (unrelated to this PR) is that both these methods have the same name but they act differently, which is odd:
- `org.eclipse.swt.internal.DPIUtil.setMonitorSpecificScaling(boolean)`: sets a system property
- `org.eclipse.swt.widgets.Display.setMonitorSpecificScaling(boolean)`: sets **and overrides** the mode.

This currently produces (AFAIK) no bug because `Display.setMonitorSpecificScaling(boolean)` is only called if the system property has been set (_e.g._ if `DPIUtil.setMonitorSpecificScaling(true)` has been called) but I think the naming of the method in `Display` could be improved in order to make more explicit what it is supposed to do and how. 

How about renaming `Display.setMonitorSpecificScaling` to `Display.forceMonitorSpecificScaling` (in a separate PR) so that it is 100% clear that this method **overrides** the system value?

No further changes required for this PR. Approved on my side.